docs site documentation on installing airgapped Kommander

vespian opened the pull request:
## Jira Ticket
https://jira.d2iq.com/browse/D2IQ-71737 is the Jira tracking this on ENG side. 
https://jira.d2iq.com/browse/D2IQ-71814 `Review Airgapped installation of Kommander` - docs team Jira

## Description of changes being made
Add information on installing airgapped Kommander. The new documentation refers heavily to airgapped konvoy install docs page, which also got updated: https://github.com/mesosphere/konvoy/pull/2171

I am not sure which kommander version it should got to. We made the changes for the current `master` branch so probably the next release.

## Checklist
- [ ] Change all affected versions, if applicable (e.g. 1.13, 2.0, 2.1).
- [ ] Test all commands and procedures, if applicable.
- [ ] Create your PR against `staging`, not `master`. 
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> ie `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

hectorj2f commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
```diff
@@ -0,0 +1,128 @@
+
+* a docker registry should already contain all the docker images required for installation and this includes also kommander images. The `konvoy_air_gapped.tar.bz2` tarball will contain the required artifacts.
+
+* connectivity with clusters that are going to be attached to the management
```
yes, the internal LBs created in the attached clusters, especially those of the federated resources, might need to be accessible from the management cluster.

hectorj2f commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
```diff
@@ -0,0 +1,128 @@
+* connectivity with clusters that are going to be attached to the management
+  cluster:
+  * both management and attached clusters should be able to connect to the docker registry
+  * management cluster should be able to connect to attached cluster's API server
```
as mentioned before, e.g. thanos needs to talk to the internal LBs exposed as internal LBs in the attached clusters.

vespian replied:
Added new bulletpoint:
```
* management cluster must connect to load balancers created by some of the addons. An example of such set up is Thanos which is part of prometheus addon
```

hectorj2f commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
```diff
@@ -0,0 +1,128 @@
+
+All nodes should meet the same minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#operating-system-and-services-for-all-nodes].
+
+## Defininig inventory file
```
typo Defininig

russurquhart1 replied:
```suggestion
## Definine the inventory file
```

hectorj2f commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
```diff
@@ -0,0 +1,128 @@
+
+The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an airgapped environment requires some additional configuration, which is described below. It is assumed that all the `cluster.yaml` changes outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] ...
+
+First of all, Kommander needs to use self-hosted charts repo that will be running on top of Konvoy cluster, as it will not be able to connect to the default one through the public Internet.
```
nit: `repository`

russurquhart1 replied:
```suggestion
1.  Ensure Kommander can use the self-hosted charts repo running on top of the Konvoy cluster. It can not connect to the default one through the public Internet.
```

cneth replied:
nit: "on top of Konvoy cluster" -> "on top of the Konvoy cluster"

hectorj2f commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
````diff
@@ -0,0 +1,128 @@
+And finally, we kommander controller needs to be reconfigured to work in an airgapped environment:
+
+```yaml
+- name: kommander
````
I personally don't like this fragmented explanation. I prefer to explain it all together.

vespian replied:
I am fine both ways. @russurquhart1 WDYT?

cmastr replied:
I agree with Hector.

vespian replied:
Done in commit `Remove explaining kommander configuration via snippets`

hectorj2f commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
````diff
@@ -0,0 +1,128 @@
+```yaml
+- name: kommander
+  values: |2
+    kommander-federation:
````
you also need to link this change too https://github.com/mesosphere/dcos-docs-site/pull/3221/files#diff-ed1b6690070eb1879058e863d81372abR103 as part of enabling airgapped for kommander

vespian replied:
Done here: https://github.com/mesosphere/dcos-docs-site/pull/3221/files#diff-ed1b6690070eb1879058e863d81372abR54

cmastr commented:
> We used to have automation for creating DOCS ticket

I don't know what this is. Yes, there should be a review ticket attached to your dev ticket for docs team to review. It is not automated for docs.

russurquhart1 commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
```diff
@@ -0,0 +1,128 @@
+
+## Control plane nodes
+
+Control plane nodes of Konvoy clusters should meet the minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#control-plane-nodes]. For 3rd-party clusters control plance requirements refer to the vendor specification.
```
```suggestion
Konvoy cluster control plane nodes must meet the minimal requirements outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#control-plane-nodes]. For 3rd-party clusters and their control plane requirements, refer to the vendor specification.
```

cneth replied:
typo:  "plance" --> "plane"

russurquhart1 commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
```diff
@@ -0,0 +1,128 @@
+
+## Worker nodes
+
+Worker nodes should meet the minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#worker-nodes].
```
```suggestion
Worker nodes must meet the minimal requirements outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#worker-nodes].
```

russurquhart1 commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
```diff
@@ -0,0 +1,128 @@
+
+## Operating system and services for all nodes
+
+All nodes should meet the same minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#operating-system-and-services-for-all-nodes].
```
```suggestion
All nodes must meet the same minimal requirements outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#operating-system-and-services-for-all-nodes].
```

russurquhart1 commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
```diff
@@ -0,0 +1,128 @@
+
+## Defininig inventory file
+
+Installing airgapped Kommander does not require any additional changes in the `inventory.yaml` file.
```
```suggestion
Installing air-gapped Kommander does not require any changes in the `inventory.yaml` file.
```

russurquhart1 commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
```diff
@@ -0,0 +1,128 @@
+
+# Configure the Kubernetes cluster
+
+The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an airgapped environment requires some additional configuration, which is described below. It is assumed that all the `cluster.yaml` changes outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] ...
```
```suggestion
The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires additional configuration. The following steps describe these changes. Ensure all the `cluster.yaml` changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation are applied.
```

russurquhart1 commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
````diff
@@ -0,0 +1,128 @@
+        chartRepo: http://konvoy-addons-chart-repo.kubeaddons.svc:8879
+```
+
+Secondly, Kommander needs to know where the private docker registry is and how to access it. The `registry_ip` variable is the ip address on which private docker registry is available, ${user} and ${password} are username and password if credentials are required.
````
```suggestion
1.  Ensure  Kommander can find and access the private Docker registry. The `registry_ip` variable is the ip address of the available private Docker registry. ${user} and ${password} are the username and password if access requires credentials.
```

russurquhart1 commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
````diff
@@ -0,0 +1,128 @@
+          docker-registry-password: 'password'
+```
+
+And finally, we kommander controller needs to be reconfigured to work in an airgapped environment:
````
```suggestion
1.  Reconfigure the Kommander controller to work in an air-gapped environment:
```

russurquhart1 commented on pages/ksphere/kommander/1.1/install-airgapped/index.md:
````diff
@@ -0,0 +1,128 @@
+              feature-gates: "Airgapped=true"
+```
+
+Putting it together, the Kommander addon configuration should look like this:
````
```suggestion
    The Kommander addon configuration should look similar to the following:
```

cneth reviewed (changes requested):
Few nits and mispellings

vespian commented:
@russurquhart1 I have ACKed all your comments/suggestions, see the commit ` Code review suggestions - Russ `
@hectorj2f I have addressed all your comments, see `Code review suggestions - Hector`
@cneth I have addressed all your comments, see `Code review suggestions - Craig`
I have also tried to address Vale lint suggestions that made sense in commit `Vale-related word-smithing`

Please have another look!

juliangieseke commented:
this PR also needs to be ported over to 1.2 🕵️

vespian commented:
> Yes, there should be a review ticket attached to your dev ticket for docs team to review. It is not automated for docs.

@cmastr https://jira.d2iq.com/browse/D2IQ-71814 `Review Airgapped installation of Kommander` I have updated the PR description as well.

vespian commented:
> this PR also needs to be ported over to 1.2 detective

@juliangieseke  Changes are for 1.2, so I moved the documentation accordingly.

russurquhart1 commented on pages/ksphere/kommander/1.2/install-airgapped/index.md:
```diff
@@ -0,0 +1,90 @@
+
+# Configure the Kubernetes cluster
+
+The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires additional configuration. Ensure all the `cluster.yaml` changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation are applied. Once this is  ...
```
```suggestion
The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires additional configuration. Ensure the `cluster.yaml` contains all the changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation. Follow this procedure:
```

russurquhart1 commented on pages/ksphere/kommander/1.2/install-airgapped/index.md:
```diff
@@ -0,0 +1,90 @@
+1.  Ensure  Kommander can find and access the private Docker registry. The `registry_ip` variable is the IP address of the available private Docker registry. ${user} and ${password} are the username and password if access requires credentials.
+1.  Reconfigure the Kommander controller to work in an air-gapped environment.
+
+The Kommander addon configuration that implements the above is as follows:
```
```suggestion
Update the Kommander addon configuration like the following:
```

russurquhart1 reviewed (approved):
had some comments below:

vespian commented:
@russurquhart1 I have ACKed all your comments/suggestions, see the commit Code review suggestions - Russ

vespian commented on pages/ksphere/kommander/1.2/install-airgapped/index.md:
```diff
@@ -0,0 +1,90 @@
+This document uses the following terms:
+- Management cluster - Konvoy cluster that is running Kommander
+- Attached cluster - Konvoy or non-Konvoy cluster that is attached to the management cluster
+* Docker registry - a registry containing all Docker images that clusters access and download during installation
```
Can you please include this change in the followup PR you will be working on?

makkes replied:
will do
